docs/samples/browser-plugin-meetings/app.js

```diff
@@ -1255,6 +1328,434 @@ function clearMediaDeviceList() {
+
+  if (!meeting) {
+    return;
+  }
+  if (!meeting.mediaProperties.webrtcMediaConnection) {
+    return;
+  }
+
+  const oldLayoutId = meeting.remoteMediaManager?.getLayoutId();
+  const newLayoutId = multistreamLayoutElm.value;
+
+  if (newLayoutId === oldLayoutId) {
+    return;
+  }
+
+  console.log(`MeetingStreams#updateMultistreamVideoLayout() :: changing layout from ${oldLayoutId} to ${newLayoutId}`);
+
+  if (newLayoutId === 'Stage') {
+    showStageSelector();
+  }
+  else if (oldLayoutId === 'Stage') {
+    hideStageSelector();
+
+    // we also undo all of the staging of members when we switch to a non-staged layout
+    resetMultistreamStage(meeting);
+  }
+
+  if (meeting.remoteMediaManager) {
+    meeting.remoteMediaManager.setLayout(newLayoutId);
+  }
+}
+
+function processNewVideoPane(meeting, paneGroupId, paneId, remoteMedia) {
+  const videoPane = allVideoPanes[paneGroupId][paneId];
+
+  videoPane.remoteMedia = remoteMedia;
+
+  remoteMedia.on('sourceUpdate', (data) => {
+    updateVideoPane(videoPane, meeting, data.state, data.memberId, `${paneGroupId}.${paneId} ${remoteMedia.id}`, 'update');
+  });
+
+  remoteMedia.on('mediaStopped', (data) => {
+    updateVideoPane(videoPane, meeting, 'no source', undefined, `${paneGroupId}.${paneId} ${remoteMedia.id}`, 'mediaStopped');
+  });
+
+  remoteMedia.on('mediaStarted', (data) => {
+    videoPane.videoEl.srcObject = data.stream;
+    updateVideoPane(videoPane, meeting, remoteMedia.sourceState, remoteMedia.memberId, `${paneGroupId}.${paneId} ${remoteMedia.id}`, 'mediaStarted');
+  });
+}
+
+function setupMultistreamEventListeners(meeting) {
+  meeting.on('media:remoteAudio:created', (audioMediaGroup) => {
+    console.log('MeetingStreams#setupMultistreamEventListeners :: got AUDIO remote media group created');
+    audioMediaGroup.getRemoteMedia().forEach((media, index) => {
+      document.getElementsByClassName('multistream-remote-audio')[index].srcObject = media.stream;
+    });
+  });
+
+  meeting.on('media:remoteVideo:layoutChanged', ({layoutId, activeSpeakerVideoPanes, memberVideoPanes}) => {
+    console.log(`MeetingStreams#VideoLayoutChanged :: got video layout changed: layoutId=${layoutId}`);
+
+    console.log('activeSpeakerVideoPanes:', activeSpeakerVideoPanes);
+    console.log('memberVideoPanes:', memberVideoPanes);
+
+    currentVideoPaneList.length = 0;
+    clearAllMultistreamVideoElements();
+    createVideoElementsForLayout(layoutId);
+
+    for (const [groupId, group] of Object.entries(activeSpeakerVideoPanes)) {
+      group.getRemoteMedia().forEach((remoteMedia, index) => processNewVideoPane(meeting, groupId, index, remoteMedia));
+    }
+
+    for (const [paneId, remoteMedia] of Object.entries(memberVideoPanes)) {
+      // staged layout is the only one we use that has memberVideoPanes defined
+      processNewVideoPane(meeting, 'stage', paneId, remoteMedia);
+    }
+  });
+
+  meeting.on('media:remoteVideoSourceCountChanged', (data) => {
```

Comment:
same here all the layout related changes should be passed on via the layout manager for the meeting and not the meeting object

Reply:
same as above